runtime-k3d/src/gen_podspec.py

```diff
@@ -0,0 +1,129 @@
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import yaml
+from yaml.loader import SafeLoader
+from typing import Optional
+
+from lib import (
+    get_services,
+    get_script_path,
+    create_nodeport
+)
+
+node_port = 30050
+
+def find_service_spec(lst, kind, name):
+    """Returns index of spec for given kind and name.
+    Args:
+        lst: The list of the template specifications.
+        kind: The kind of the pod.
+        name: The name of the pod.
+    """
+    return [i for i, elem in enumerate(lst) if elem['kind'] == kind and elem['metadata']['name'] == name]
+
+def init_template(templates):
+    """Initializes podspecs list with non generated podspecs.
+    Args:
+        templates: The list of the template specifications.
+    """
+    template = []
+    template.append(templates[find_service_spec(templates, 'Pod', 'bash')[0]])
+    template.append(templates[find_service_spec(templates, 'ConfigMap', 'feeder-config')[0]])
+    template.append(templates[find_service_spec(templates, 'PersistentVolume', 'pv-volume')[0]])
+    template.append(templates[find_service_spec(templates, 'PersistentVolumeClaim', 'pv-claim')[0]])
+    
+    return template
+
+def create_podspec(templates, service_spec):
+    """Creates podspec for given service specification.
+
+    Args:
+        templates: The list of the template specifications.
+        service_spec: The specification of the service.
+    """
+    global node_port
+    
+    service_id = service_spec["id"]
+    
+    container_image = None
+    service_port = None
+    env_vars = dict[str, Optional[str]]()
+    port_forwards = []
+    mounts = []
+    args = []
+    pods=[]
+
+    for config_entry in service_spec["config"]:
+        if config_entry["key"] == "image":
+            container_image = config_entry["value"]
+        elif config_entry["key"] == "env":
+            pair = config_entry["value"].split("=", 1)
+            env_vars[pair[0].strip()] = None
+            if len(pair) > 1:
+                env_vars[pair[0].strip()] = pair[1].strip()
+        elif config_entry["key"] == "port":
+            service_port = config_entry["value"]
+        elif config_entry["key"] == "no-dapr":
+            no_dapr = config_entry["value"]
+        elif config_entry["key"] == "arg":
+            args.append(config_entry["value"])
+        elif config_entry["key"] == "port-forward":
+            port_forwards.append(config_entry["value"])
+        elif config_entry["key"] == "mount":
+            mounts.append(config_entry["value"])
```

Comment:
this looks the same like in the helm, could we somehow create a shared function in the lib for it?